Fix multiple notifications arriving for imports in edge cases

peppy opened the pull request:
I've noticed this a few times, and caught it once earlier today (in log form). This has got to be the reason for it.

```
[database] 2022-11-02 03:48:45 [verbose]: [8fa77] Import successfully completed!
[runtime] 2022-11-02 03:48:45 [verbose]: ⚠️ Imported Unlike Pluto - Everything Black (feat. Mike Taylor) (CoolGuyRift)! Click to view.
[runtime] 2022-11-02 03:48:45 [verbose]: ⚠️ Imported Unlike Pluto - Everything Black (feat. Mike Taylor) (CoolGuyRift)! Click to view.
```

Probably closes https://github.com/ppy/osu/issues/20875.

peppy committed:
Fix multiple notifications arriving for imports in edge cases

smoogipoo reviewed (dismissed):
The threading here is a little bit dodgy and feels like we're pointing a gun at our foot, but I suppose that's a separate issue that we'll address when the bullet is fired. As it is this doesn't change logic, just makes it thread-safe...

smoogipoo committed:
Use Interlocked.Exhange() instead

Increment isn't correct since it returns the post-incremented value. It
also always increments.

smoogipoo committed:
Fix inverted condition
